### ISSUE TITLE
False "Cycle lane on this side of the street"

### ISSUE THREAD

**Issue by shadow-identity** [cyclosm/cyclosm-cartocss-style#578]
Hey, thank you for the great render!

I just found that in case of 
```
cycleway: opposite_lane
highway: residential
lanes: 1
oneway: yes
oneway:bicycle: no
```
it renders incorrectly, with cycle lanes on the both sides. While there is only one cycle lane (going opposite). There is no cycle lane in the direction of the line (and it is not described in the tags).

![image](https://user-images.githubusercontent.com/1223112/120218393-f6888a80-c239-11eb-93df-4c3117aa9408.png)

The link to the point on map: https://www.cyclosm.org/#map=19/47.06975/15.43023/cyclosm_live

**Comment by Phyks** [comment]
Hi,

Thanks for the feedback and the issue! `cycleway=opposite_lane` is somehow an imprecise tagging, as it does not indicate which lane the cycleway is expected to be.

Most of the case it will be on the expected side of driving (right when riding countraflow for most of the world, left for some countries), but there are many exceptions around the globe. Therefore, we decided to render it on both sides of the road to favour more explicit tagging schemes (lateralized).

We used to have "shark teeths" markers for the cycleways to indicate direction. See screenshot in https://github.com/cyclosm/cyclosm-cartocss-style/issues/260#issuecomment-570613739. That way, it was clear for everyone that cycleways on both side of the street were actually going in the same direction and it was giving an indication that these double cycleways were likely artifacts.

Seems we lost it in latest version, we'll have to look into it.

**Comment by Florimondable** [comment]
"shark teeths" has been remoed in an earlier version for optimisation purpose, you were ok with that :)

**Comment by Phyks** [comment]
True, I remember this now, although I cannot find back the PR about this.

I'm still in favor of removing shark teeths for optimization purposes, but looking at it again I think they might be missing in this specific case of non-lateralized cycleways. Maybe it could be worth restoring them only for this specific case?

**Comment by Phyks** [comment]
@shadow-identity I don't understand the tagging scheme in your latest example.

If the `cycleway` and `cycleway:left` tags describe the same physical infrastructure, they should IMO be merged into a single `cycleway:left=opposite_lane` (see https://wiki.openstreetmap.org/wiki/Key:cycleway:left), or even better a `cycleway:left=lane; cycleway:left:oneway=-1`.
